### PR TITLE
[Serializer] Remove deprecation layer

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | N/A
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

File: src/Symfony/Component/Serializer/Annotation/Context.php [hunk 1]
```diff
@@ -25,58 +25,30 @@
...
-        $normalizationContext = $options['normalizationContext'] ?? $normalizationContext;
-        $denormalizationContext = $options['denormalizationContext'] ?? $denormalizationContext;
-
-        foreach (compact(['context', 'normalizationContext', 'denormalizationContext']) as $key => $value) {
-            if (!\is_array($value)) {
-                throw new InvalidArgumentException(sprintf('Option "%s" of annotation "%s" must be an array.', $key, static::class));
-            }
-        }
-
+    public function __construct(
```
Comment (OskarStark) [inline src/Symfony/Component/Serializer/Annotation/Context.php:35]: Should be on one line

File: src/Symfony/Component/Serializer/Annotation/DiscriminatorMap.php [hunk 2]
```diff
@@ -26,41 +26,19 @@
...
-        if (\is_array($typeProperty)) {
-            trigger_deprecation('symfony/serializer', '5.3', 'Passing an array as first argument to "%s" is deprecated. Use named arguments instead.', __METHOD__);
-
-            $mapping = $typeProperty['mapping'] ?? null;
-            $typeProperty = $typeProperty['typeProperty'] ?? null;
-        } elseif (!\is_string($typeProperty)) {
-            throw new \TypeError(sprintf('"%s": Argument $typeProperty was expected to be a string or array, got "%s".', __METHOD__, get_debug_type($typeProperty)));
-        }
-
+    public function __construct(
```
Comment (OskarStark) [inline src/Symfony/Component/Serializer/Annotation/DiscriminatorMap.php:28]: Same here

File: src/Symfony/Component/Serializer/Annotation/Context.php [hunk 3]
```diff
@@ -25,58 +25,30 @@
...
+        private array $context = [],
+        private array $normalizationContext = [],
+        private array $denormalizationContext = [],
+        string|array $groups = []
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Serializer/Annotation/Context.php]: let's always add the trailing coma when using this syntax

**Review by nicolas-grekas — COMMENTED** [review]
please rebase so that we can see green tests :)

**Comment by derrabus** [comment]
Fabbot failure is the usual false positive, AppVeyor failure is unrelated. The remaining tests are green.

File: src/Symfony/Component/Serializer/Annotation/Context.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/Serializer/Annotation/Context.php:35]: when using constructor property promotion, I think we should use several lines actually
we do so already for attributes

**Comment by nicolas-grekas** [comment]
Thank you @derrabus.

**Pull request merged**